sycl/include/CL/sycl/xilinx/fpga/kernel_properties.hpp

```diff
@@ -25,6 +25,33 @@
 __SYCL_INLINE_NAMESPACE(cl) {
 
 namespace sycl::xilinx {
+
+namespace literal {
+
+#ifdef __SYCL_DEVICE_ONLY__
+
+namespace  {
+template<char...Chars>
+constexpr const char FixedString[] = {Chars...};
+}
+
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wgnu-string-literal-operator-template"
```

Comment:
What is the problem here?
At least mention it in a comment.
I think this should be compiled in C++20 mode anyway.

Reply:
And by the way, as usual, add a lot of comments all over the place.

Reply:
this is not using C++20 but a compiler extension from clang and gcc, because this is only needed for device code using clang compiler extentions is fine. but this is to silence the warning about that compiler extension.

Reply:
This is a compiler extension not C++20 so this doesn't need C++20

Reply:
All this requires even more comments and to be moved into a separate file higher in the hierarchy, the equivalent of `include/triSYCL/detail/...`.